stats/sexbound/sexbounddefeat.lua

```diff
@@ -0,0 +1,568 @@
+		_sexboundConfig    = nil
+  }, SexboundDefeat_mt)
+
+  	_self:setIsDefeated(false)
+
+	-- Load the Sexbound Defeat config settings
+	_self._config = _self:loadConfig()
+	_self:validateConfig(_self._config)
+
+	-- Initialize message handlers
+	_self:initMessageHandlers()
+
+	-- Set the default timeout as configured in the settings
+	_self:_initTimeout()
+
+	-- Set "Can use sexbound UI while defeated." state as a status for use with other scripts
+	status.setStatusProperty("can_use_sex_ui_defeated", _self._config.defeatedPlayersCanUseSexUI)
+
+	-- Enable canBeDefeated and canDefeatOthers on non-players. Additional conditions could be added.
+	if _self.entityType ~= "player" then
+		status.setStatusProperty("canBeDefeated", true)
+		status.setStatusProperty("canDefeatOthers", true)
+	end
+	return _self
+end
+
+function SexboundDefeat:initMessageHandlers()
+	message.setHandler("SexboundDefeat:Breakout", function(_,_,args)
+		self._timer = self._timeout
+	end)
+	message.setHandler("SexboundDefeat:TargetedBy", function(_,_,entityId)
+		-- Send hostile entity a message to add self to their hostile entity table
+		if status.statusProperty("canDefeatOthers") then
+			world.sendEntityMessage(entityId, "SexboundDefeat:InCombatWith", self._entityId)
+		end
+		local found = false
+		for i, v in ipairs(self._hostileEntities) do
+			if v[1] == entityId then found = true end
+		end
+		if not found then
+			self:retrieveSexboundConfig(function(result)
+				-- Run function that adds entity with their sexbound config into the hostile entity table
+				return self:handleRetrieveSexboundConfigSuccess(result,entityId)
+			end, function()
+				return
+			end,
+			entityId)
+		end
+	end)
+	-- Ends the hostile entity "handshake". Does same as "TargetedBy" but will not an additional message.
+	message.setHandler("SexboundDefeat:InCombatWith", function(_,_,entityId)
+		local found = false
+		for i, v in ipairs(self._hostileEntities) do
+			if v[1] == entityId then found = true end
+		end
+		if not found then
+			self:retrieveSexboundConfig(function(result)
+				return self:handleRetrieveSexboundConfigSuccess(result,entityId)
+			end, function()
+				return
+			end,
+			entityId)
+		end
+	end)
+	-- Sets the entity's status to untransform at the earliers opportunity
+	message.setHandler("SexboundDefeat:Untransform", function(_,_)
+		self:setTimer(self._timeout - 3)
+	end)
+end
+
+-- Update Loop
+function SexboundDefeat:update(dt, oldUpdate)
+	self._promises:update()
+
+	if self:isTransformed() then
+    	self._timer = self._timer + dt
+    	-- Untransform the NPC when the timer runs out
+    	if self._timer >= self._timeout or not self:isStunned() then
+        	self:setIsDefeated(false)
+	    	self:setIsTransformed(false)
+	    	self:setTimer(0)
+        	self:untransform()
+			status.removeEphemeralEffect("sexbound_defeat_stun")
+    	end
+	end
+
+	-- Return when is defeated to prevent the default update loop from running
+	if self:isDefeated() then return end
```

Comment:
Are we sure we need to completely omit the normal update loop? That would prevent pregnancies from updating on players, among other things.

Reply:
That is a remnant from old defeat. I'm not sure why we'd prevent the normal update loop. I'll remove it and see what happens.